### DISCUSSION TITLE
Lights always on after restart and power cycle

### DISCUSSION THREAD

**Discussion by joe-sydney** [arendst/Tasmota#14730] [Q&A] [answered]
Hi all, I must be missing something, but it seems like my lights are always on after restart and power cycle.

Regarding power cycle, I tried with PowerOnState both 0 and 1. When set to 0, I had expected the light to be off once it gets power.

Regarding restart (either via the web gui, or "restart 1" command), I had expected the light to be on or off depending on its state before the restart. Setoption0 is on.

But in all situations, the light turns on. Note, I tested on 2 separate lights. 

Any pointers would be much appreciated!

**Comment by sfromis** [comment, marked as answer]
May also depend on your `SaveData` setting, and if you have any rules or other automation. Watching the console (after setting Weblog 3) will often be quite informative about why things happen. If your light is afflicted with a TuyaMCU, that thing may independently have lights on at powerup, to facilitate lights working behind an old-fashioned wall switch.

**Reply by joe-sydney** [reply]
Indeed, I had a rule that initialised the ct and brightness (dimmer), but I forgot that although the rule did not explicitly turn the power on, by default changing the dimmer value turned the power on. Setting setoption 20 solved that.

Thanks for the reminder!
